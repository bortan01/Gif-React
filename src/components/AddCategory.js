import React, { useState } from 'react'
import PropTypes from 'prop-types';

export const AddCategory = ({ setCategories }) => {
   const [inputValue, setInputValue] = useState("");
   const handleInputChance = (e) => {
      setInputValue(e.target.value);
   }
   const handleSubmit = (e) => {
      e.preventDefault();
      console.log("submit");
      if (inputValue.trim().length > 2) {
         setCategories((categorias) => [inputValue, ...categorias]);
         setInputValue('');
      }
   }
   return (
      <form onSubmit={handleSubmit}>
         <input
            type="text"
            value={inputValue}
            onChange={(e) => { handleInputChance(e) }}
         />
      </form>
   )
}

AddCategory.propTypes = {
   setCategories: PropTypes.func.isRequired
}