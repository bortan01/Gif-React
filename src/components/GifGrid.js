import React from 'react'

export const GifGrid = ({category}) => {

   const getGifs =async ()=>{
      const url ='https://api.giphy.com/v1/gifs/search?q=goku&limit=10&api_key=R4qK3NSI73RXvjZSdZ8hCDYXSfEq0xSj';
      const respuesta = await fetch(url);
      const {data} = await respuesta.json();

      const gifs = data.map((img)=>{
         return {
            id : img.id,
            url : img.images?.downsized_medium.url,
            title : img.title
         }
      });
      console.log(gifs);
   }
   getGifs();
   return (
      <div>
         <h3>{category}</h3>
      </div>
   )
}
