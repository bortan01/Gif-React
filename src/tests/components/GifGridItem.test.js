import React from 'react';
import { shallow } from "enzyme";
import { GifGridItem } from "../../components/GifGridItem";

describe('pruebas en GifGridItem', () => {
   const title = 'Un titulo';
   const url = 'https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcRTXlets_YMDLRWRdARL10-uXyzSbK3v7rTgw&usqp=CAU';
   const wrapper = shallow(<GifGridItem title={title} url={url} />);

   test('debe de mostrar el componente correctamente ', () => {
      expect(wrapper).toMatchSnapshot();
   });
   test('debe de tener un parrafo con el titulo ', () => {
      const p = wrapper.find('p');
      expect(p.text().trim()).toBe(title);
   });
   test('debe de tener la imagen igual a la url y alt de prots ', () => {
      const img = wrapper.find('img');
      expect(img.prop('src')).toBe(url);
      expect(img.prop('alt')).toBe(title);
   });
   test('debe de tener animate__fadeOut ', () => {
      const div = wrapper.find('div');
      expect(div.prop("className").includes("animate__fadeOut")).toBe(true);
   });


});
