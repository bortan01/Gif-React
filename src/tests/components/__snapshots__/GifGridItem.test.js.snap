// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en GifGridItem debe de mostrar el componente correctamente  1`] = `
<div
  className="card animate__fadeOut"
>
  <img />
  <p />
</div>
`;
